tiledb/cloud/client.py

```diff
@@ -375,148 +372,154 @@ def find_organization_or_user_for_default_charges(user):
     return namespace_to_charge
 
 
+class RetryMode(enum.Enum):
+    DEFAULT = "default"
+    FORCEFUL = "forceful"
+    DISABLED = "disabled"
+
+    def maybe_from(v: "RetryOrStr") -> "RetryMode":
+        if isinstance(v, RetryMode):
+            return v
+        return RetryMode(v)
+
+
+RetryOrStr = Union[RetryMode, str]
+
+
+_RETRY_CONFIGS = {
+    RetryMode.DEFAULT: urllib3.Retry(
+        total=10,
+        backoff_factor=0.25,
+        status_forcelist=[503],
+        allowed_methods=[
+            "HEAD",
+            "GET",
+            "PUT",
+            "DELETE",
+            "OPTIONS",
+            "TRACE",
+            "POST",
+            "PATCH",
+        ],
+        raise_on_status=False,
+        # Don't remove any headers on redirect
+        remove_headers_on_redirect=[],
+    ),
+    RetryMode.FORCEFUL: urllib3.Retry(
+        total=10,
+        backoff_factor=0.25,
+        status_forcelist=[400, 500, 501, 502, 503],
+        allowed_methods=[
+            "HEAD",
+            "GET",
+            "PUT",
+            "DELETE",
+            "OPTIONS",
+            "TRACE",
+            "POST",
+            "PATCH",
+        ],
+        raise_on_status=False,
+        # Don't remove any headers on redirect
+        remove_headers_on_redirect=[],
+    ),
+    RetryMode.DISABLED: False,
+}
+
+
+# Type of the callback function we pass the response UUID.
+IDCallback = Callable[[Optional[uuid.UUID]], Any]
+
+
+def send_udf_call(
+    api_func: Callable[..., urllib3.HTTPResponse],
+    api_kwargs: Dict[str, Any],
+    decoder: results.AbstractDecoder,
+    *,
+    id_callback: IDCallback,
+    store_results: bool,
+) -> results.Response:
+    """Sends an API request and returns the results in the form of a Response."""
+    try:
+        http_response = api_func(_preload_content=False, **api_kwargs)
+    except rest_api.ApiException as exc:
+        id_callback(results.extract_task_id(exc))
+        raise tiledb_cloud_error.check_exc(exc) from exc
```

Comment:
In the previous code, `raise X from None` was specifically used to avoid including the whole local rest call stack for remote errors; it makes them unreadable. AFAICT `check_exc` will already raise `from exc` when appropriate: https://github.com/TileDB-Inc/TileDB-Cloud-Py/blob/81c07150fd70f60a674794559bc20ae42678048e/tiledb/cloud/tiledb_cloud_error.py#L29

Reply:
Fixed.